### PR TITLE
Updating key values of device properties to dict

### PR DESCRIPTION
**Pull request by samtwum**
The fandango parser assigns a dict to the key values for the different properties in the form
 {
'property-name': {
    'DefaultPropValue': 'list-of-strings',
    'name': 'property-name',
     'type': 'data-type'},
 }
This will eliminate the current errors encountered when the launcher is run. Then the task in ticket number below can be addressed.

JIRA ticket: [CB-2917](https://skaafrica.atlassian.net/browse/CB-2917)

### CODE REVIEW

File: tango_simlib/utilities/fandango_json_parser.py [hunk 1]
```diff
@@ -82,22 +82,50 @@ def preprocess_attribute_types(self, attribute_data):
...
                         getattr(AttrDataFormat, attr_prop_value))
 
         for attr, max_config in max_dim.items():
             attribute_data[attr].update(max_config)
 
-        self._device_attributes.update(attribute_data)           
+        self._device_attributes.update(attribute_data)
+
+    def update_property_data(self, property_data):
+        """Update key values to a dict with keys 'DefaultPropValue','name' and 'type'
```
Comment (SKAJohanVenter) [inline tango_simlib/utilities/fandango_json_parser.py]: Please add a `Parameters` section describing `property_data` in the docstring.
See https://docs.google.com/document/d/1aZoIyR9tz5rCWr2qJKuMTmKp2IzHlFjrCFrpDDHFypM/edit#heading=h.9lg6cmlfdx6i

File: tango_simlib/utilities/fandango_json_parser.py [hunk 2]
```diff
@@ -82,22 +82,50 @@ def preprocess_attribute_types(self, attribute_data):
             for attr_prop, attr_prop_value in attr_config.items():
                 if attr_prop == 'data_type':
                     attr_config[attr_prop] = getattr(CmdArgType, attr_prop_value)
+                    # ensuring READ_WRITE is assigned to 'writable' key for all DevEnum
```
Comment (lvdheever) [inline tango_simlib/utilities/fandango_json_parser.py]: Why is it necessary to force DevEnums to 'writable'. I can see cases where you may want to enumerate a state (e.g. is a device a SIM device or a REAL device) but not want it to be writable. Is there something in the TANGO implementation (or ours) that requires this?

File: tango_simlib/utilities/fandango_json_parser.py [hunk 3]
```diff
@@ -82,22 +82,50 @@ def preprocess_attribute_types(self, attribute_data):
...
+        for prop, prop_val in property_data.items():
+            prop_data[prop] = {
+                'DefaultPropValue': prop_val,
+                'name': prop,
+                'type': 'VarStringArray'
+                }
+
+        for prop, config in prop_data.items():
+            property_data[prop] = config
+
```
Comment (lvdheever) [inline tango_simlib/utilities/fandango_json_parser.py]: You can possibly do this in a more pythonic way, and not sure you need the 'name' item as it already is the key (unless there is some compatiblity reason for this).
Something like this:
'''
prop_data = [(prop, dict([('DefaultPropValue', prop_val),
                                          ('name', prop),
                                          ('type', 'VarStringArray')]))
                    for prop, prop_val in property_data.items()]
property_data = dict(prop_data)
```

File: tango_simlib/utilities/fandango_json_parser.py (hunk 2 above)
Comment (ajoubertza) [reply tango_simlib/utilities/fandango_json_parser.py]: @lvdheever According to @samtwum the file exported using Fandango doesn't accurately report whether enum attributes are read only, write only, or both.  For this reason, the most general option was selected.  It is something that should be fixed in the Fandango exporter.

Comment (samtwum) [reply tango_simlib/utilities/fandango_json_parser.py]: @lvdheever Just an addition to what @ajoubertza has explained...
Some of the DevEnum type attributes have 'WT_UNKNOWN' as the value registered for the attribute property 'writable' after exporting the data from the running device in the VM.
So the parser forces all values of the 'writable' attribute property to 'READ_WRITE'.

File: tango_simlib/utilities/fandango_json_parser.py [hunk 4]
```diff
@@ -82,23 +82,68 @@ def preprocess_attribute_types(self, attribute_data):
...
+        e.g.
+            {
+                '<property-name>': '<list-of-strings>',
+                '<property-name>': '<list-of-strings>'
+            }
 
+        property_data is reformatted to the format below
+            {
+                '<property-name>': {
+                    'DefaultPropValue': '<list-of-strings>',
```
Comment (kmadisa) [inline tango_simlib/utilities/fandango_json_parser.py]: - [ ] Update the docstring in `def get_device_properties_metadata(...`.

**Pull request merged**